### DISCUSSION TITLE
Errors when trying to run unms container

### DISCUSSION THREAD

**Discussion by pgrudzin** [Nico640/docker-unms#86] [General]
Hello @Nico640 

I'm trying to run unms container using podman on Ubuntu 22.04 LTS.

Since I want unms container to have it's own IP, I'm using macvlan networking. This is the command I perform to run the container:

`podman network create -d macvlan -o parent=eth0 --subnet 192.168.6.0/24 --gateway 192.168.6.1 podnet`

`podman volume create uisp_config`

`podman run -d --name=uisp --network=podnet --ip=192.168.6.9 --mac-address=5e:fa:04:c8:89:7c -e TZ=Europe/Warsaw -v uisp_config:/config --restart=unless-stopped --hostname=uisp nico640/docker-unms`

While same command works perfectly with e.g. my pihole containers, I'm not able to get unms container started correctly. Not sure where the root cause lies though. 

Attaching the complete execution log, at least I could see the following errors repeating in there, although not sure if they are indeed a real issue or not:

`Restoring backups and/or running migrations`
`/usr/local/bin/yarn: exec: line 20: node: Operation not permitted`
`2023-03-07 22:59:12.596 CET [1249] FATAL:  role "root" does not exist`

[run.log](https://github.com/Nico640/docker-unms/files/10918286/run.log)

The end result is that container is up, but when trying to access UISP web page, it opens and says "initiating, please wait" and after some few mins waiting gives error and stops.

I would very much appreciate getting some insight into what is causing this error and how to solve it.

BRs
Pawel

**Comment by pgrudzin** [comment]
Hello @Nico640 
any chance for your feedback please? :-)
BRs
Pawel

**Comment by Nico640** [comment]
Sorry, missed this. The issue is definitely this: `/usr/local/bin/yarn: exec: line 20: node: Operation not permitted`
Not sure why this is happening, but it seems to be coming from line 161 in /usr/local/bin/docker-entrypoint.sh, which is `yarn prestart`.
Can you try running `yarn prestart` inside the container as the unms user and see what happens? Also try running `node`. Both yarn and node are located / linked in `/usr/local/bin/`, check if they are both executable. I'm not familiar with podman, but maybe this is some kind of permission issue.

**Reply by pgrudzin** [reply]
Thanks for showing the interest and troubleshooting hints! This looks odd:

`uisp:~$ id`
`uid=911(unms) gid=911(unms) groups=911(unms)`

`uisp:~$ node`
`-ash: node: Operation not permitted`

`uisp:~$ ls -l /usr/local/bin/node`
`-rwxr-xr-x 1 root root 80420080 Jan  3  2022 /usr/local/bin/node`

`uisp:~$ yarn prestart`
`/usr/local/bin/yarn: exec: line 20: node: Operation not permitted`

`uisp:~$ ls -l /usr/local/bin/yarn`
`lrwxrwxrwx 1 root root 27 Jan  3  2022 /usr/local/bin/yarn -> /opt/yarn-v1.22.15/bin/yarn`

`uisp:~$ ls -l /opt/yarn-v1.22.15/bin/yarn`
`-rwxr-xr-x 1 root root 1025 Sep 30  2021 /opt/yarn-v1.22.15/bin/yarn`


I'm not sure if this is some podman "speciality", I will try to google. In case you have some further clue what can be wrong or what else to check, I'll be more than happy to get your feedback.

**Reply by pgrudzin** [reply]
For a test I've tried other command from `/usr/local/bin`, and e.g. php worked fine with same file permissions:

`uisp:~$ php -v`
`PHP 7.4.26 (cli) (built: Jan 31 2023 19:47:24) ( NTS )`
`Copyright (c) The PHP Group`
`Zend Engine v3.4.0, Copyright (c) Zend Technologies`
`    with Zend OPcache v7.4.26, Copyright (c), by Zend Technologies`
`uisp:~$ which php`
`/usr/local/bin/php`

`uisp:~$ ls -l /usr/local/bin/php`
`-rwxr-xr-x 1 root root 14434760 Jan 31 20:51 /usr/local/bin/php`

**Reply by Nico640** [reply]
Hmm, yeah, looks like only node is not executing for some reason... is it executable when running under root?

**Reply by pgrudzin** [reply]
Same result with root - operation not permitted

**Reply by Nico640** [reply]
Try executing node in a `node:16.13.1-alpine3.15` container, which is the base image the UISP container uses. That would rule out anything UISP related. If that still doesn't work, try executing the container in privileged mode. If it works in privileged mode, then it's probably something related with container permissions

**Reply by pgrudzin** [reply]
I didn't try with alpine container, but using `--privileged` solved the problem. Of course at the cost of lowered security :-)
Anyway, at least I know which direction to look at for more proper solution.
Thanks for your time and hints which allowed to at least see UISP in action :-)
Best regards
